src/features/currentTodo.ts

```diff
@@ -1,10 +1,18 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
+import { RootState } from '../app/store';
 
 const initialState = null as Todo | null;
 
 export const currentTodoSlice = createSlice({
   name: 'currentTodo',
   initialState,
-  reducers: {},
+  reducers: {
+    setTodo: (_, { payload }: PayloadAction<Todo | null>) => payload,
+    clearTodo: () => initialState,
+  },
 });
+
+export const { setTodo, clearTodo } = currentTodoSlice.actions;
+
+export const selectCurrentTodos = (state: RootState) => state.todo;
```

Comment:
The selector `selectCurrentTodos` should reference `state.currentTodo` instead of `state.todo` to correctly select the current todo from the state.